packages/reaction-core/server/methods/cart.js

```diff
@@ -344,13 +464,21 @@ Meteor.methods({
    * @todo:  Partial order processing, shopId processing
    * @todo:  Review Security on this method
    * @param {String} cartId - cartId to transform to order
+   * @param {String} sessionId - current client session id. It is needed to
+   * create new cart within `cart/createCart` method
    * @return {String} returns orderId
    */
-  "cart/copyCartToOrder": (cartId) => {
+  "cart/copyCartToOrder": function (cartId, sessionId) {
     check(cartId, String);
-    let cart = ReactionCore.Collections.Cart.findOne(cartId);
-    let order = _.clone(cart);
-    let user;
+    check(sessionId, String);
+
+    const cart = ReactionCore.Collections.Cart.findOne(cartId);
+    // security check
+    if (cart.userId !== this.userId) {
+      throw new Meteor.Error(403, "Access Denied");
+    }
+    const order = Object.assign({}, cart);
+
```

Comment:
As @zenweasel says "Boom". Your right. We could just copy `sessionId` from previous cart to new one. It should be fine in case then anonymous make payment. I will try this tomorrow